bin/faucet/src/faucet/mod.rs

```diff
@@ -242,13 +243,17 @@ impl Faucet {
         let tx_executor =
             Rc::new(TransactionExecutor::new(data_store.clone(), Some(authenticator.clone())));
 
+        let coin_seed: [u64; 4] = random();
+        let rng = RpoRandomCoin::new(coin_seed.map(Felt::new));
+
         Ok(Self {
             data_store,
             id,
             prior_state: VecDeque::new(),
             tx_prover,
             tx_executor,
             account_interface,
+            rng: Arc::new(Mutex::new(rng)),
```

Comment:
I'm just wondering if we prefer to take the compute hit on creating a new `RpoRandomCoin` for every batch request instead of locking over the relevant section given this is part of the faucet RPC/API stack.
```
        let p2id_notes = {
            let mut rng_guard = self.rng.lock().await;
            P2IdNotes::build(self.faucet_id(), requests, &mut rng_guard)?
        };
```

Only significant if we expect that section to take like an order of magnitude or more than generating the `RpoRandomCoin`. Probably not worth spending time benchmarking etc, just wondering.

Reply:
Yeah, we briefly discussed this as well. Initializing an `RpoRandomCoin` from `thread_rng` does not look like it's trivial but it's probably less overhead than keeping a global lock (not sure by how much, though). We could also keep `thread_local` coins and take those for less locking overhead, but seems overkill. I don't have too much of a strong opinion as this depends a lot on traffic patterns (I think we can expect a relatively low number of concurrent request to hit this point) and it's hard to tell without measuring but I think I lean towards instantiating a new coin each thread.

If we keep this it might be a good idea to emit a trace of how long it took to acquire the lock maybe.

Reply:
I think if the purpose of the `Arc<Mutex<` is for performance, I would remove it. If we are concerned, we could add a TODO or NOTE about performance overhead of the relevant section for each batch request.

I prefer to not use the `Arc<Mutex<` because:
1. The use case here is not one of state sharing; and
2. The performance improvement is speculative or at least likely nominal.

I also suspect that the faucet will never be compute bound (so the random coin generation is fine to repeat).

Just my 2 cents. Feel free to proceed how you prefer.